include/gnc_environment.hpp

```diff
@@ -0,0 +1,68 @@
+//
+// include/gnc_environment.hpp
+// PSim
+//
+// Contributors:
+//   Kyle Krol  kpk63@cornell.edu
+//
+// Pathfinder for Autonomous Navigation
+// Space Systems Design Studio
+// Cornell Univeristy
+//
+
+#ifndef PAN_PSIM_INCLUDE_GNC_ENVIRONMENT_HPP_
+#define PAN_PSIM_INCLUDE_GNC_ENVIRONMENT_HPP_
+
+#include "lin.hpp"
+
+namespace gnc {
+namespace env {
+
+/** @fn earth_attitude
+ *  @param[in]  t Time in seconds since the PAN epoch.
+ *  @param[out] q Quaternion to transform vectors from ECI to ECEF. */
+void earth_attitude(double t, lin::Vector4d &q);
+
+/** @fn earth_attitude
+ *  @param[in]  t Time in seconds since the PAN epoch.
+ *  @param[out] q Quaternion to transform vectors from ECI to ECEF. */
+void earth_attitude(double t, lin::Vector4f &q);
+
+/** @fn earth_angular_rate
+ *  @param[in]  t Time in seconds since the PAN epoch.
+ *  @param[out] w Angular rate to transform vectors from ECI to ECEF. */
+void earth_angular_rate(double t, lin::Vector3d &w);
+
+/** @fn earth_angular_rate
+ *  @param[in]  t Time in seconds since the PAN epoch.
+ *  @param[out] w Angular rate to transform vectors from ECI to ECEF. */
+void earth_angular_rate(double t, lin::Vector3f &w);
+
+/** @fn sun_vector
+ *  @param[in]  t Time in seconds since the PAN epoch.
+ *  @param[out] s Normalized vector from Earth the Sun in the ECI frame. */
+void sun_vector(double t, lin::Vector3d &s);
```

Comment:
I'm pretty sure we don't need a double version of this.

Reply:
I'm going to leave it I think simply because we need to do the calculation in doubles up until the quaternion is normalized. Otherwise we'd risk losing a lot of precision once `t` gets large.

Reply:
Got rid of the double version after realizing we'd need to be in orbit for 100's of years for accuracy to become a problem. By that point the approximations made by this function would already fall apart anyway.